### DISCUSSION TITLE
Remove support for infix functions

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#7874] [Ideas]
The proposal is to remove support for infix function calls:

```scala
let result = 123 `sum` 456
```

The reason is twofold: (a) it makes function calls more uniform (less syntax), and (b) it frees up tilde for use in escaping keywords, e.g.:

```scala
let o = 123ii; // BigInt object
o.`and`(o) // and is normally a keyword, but here it is escaped by ``.
```

**Comment by JonathanStarup** [comment]
There is also a conflict between infix and `|>` (point-free?)
```
def lessThan(x: Int32, y: Int32): Bool = x < y

42 |> lessThan(30)
30 `lessThan` 42
```
although `|>` also conflicts with normal application in binary operations

**Reply by mlutze** [reply]
This is a major concern. A user will certainly tend to write the `lessThan` function like that.
